From 2648449dcd82aedc60593f551866256a96cf2fc7 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 27 May 2019 09:55:19 +0800
Subject: [PATCH] sysnetwork: fix avc denials for net_conf_t

Fix:
type=AVC msg=audit(1558599938.333:114): avc:  denied  { read } for
pid=1575 comm="local" name="resolv.conf" dev="vda" ino=30304
scontext=system_u:system_r:postfix_local_t:s0-s15:c0.c1023
tcontext=system_u:object_r:net_conf_t:s0 tclass=lnk_file permissive=1

type=AVC msg=audit(1558599949.992:115): avc:  denied  { read } for
pid=222 comm="snmptrapd" name="resolv.conf" dev="vda" ino=30304
scontext=system_u:system_r:snmpd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:net_conf_t:s0 tclass=lnk_file permissive=1

type=AVC msg=audit(1558599964.617:116): avc:  denied  { read } for
pid=172 comm="systemd-timesyn" name="resolv.conf" dev="vda" ino=30304
scontext=system_u:system_r:ntpd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:net_conf_t:s0 tclass=lnk_file permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/sysnetwork.fc | 1 +
 policy/modules/system/sysnetwork.if | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/policy/modules/system/sysnetwork.fc b/policy/modules/system/sysnetwork.fc
index da144fc..0e9f74d 100644
--- a/policy/modules/system/sysnetwork.fc
+++ b/policy/modules/system/sysnetwork.fc
@@ -34,6 +34,7 @@ ifdef(`distro_redhat',`
 /etc/sysconfig/networking(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
 /etc/sysconfig/network-scripts(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
 /run/systemd/resolve/resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
+/run/systemd/resolve/stub-resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
 /etc/resolv-conf\.systemd	-l	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/resolv\.conf	-l	gen_context(system_u:object_r:net_conf_t,s0)
 ')
diff --git a/policy/modules/system/sysnetwork.if b/policy/modules/system/sysnetwork.if
index c9d5c80..31d807d 100644
--- a/policy/modules/system/sysnetwork.if
+++ b/policy/modules/system/sysnetwork.if
@@ -347,6 +347,7 @@ interface(`sysnet_read_config',`
 
 	files_search_etc($1)
 	allow $1 net_conf_t:file read_file_perms;
+	allow $1 net_conf_t:lnk_file read_file_perms;
 
 	ifdef(`distro_debian',`
 		files_search_pids($1)
@@ -381,6 +382,7 @@ interface(`sysnet_dontaudit_read_config',`
 	')
 
 	dontaudit $1 net_conf_t:file read_file_perms;
+	dontaudit $1 net_conf_t:lnk_file read_file_perms;
 ')
 
 #######################################
-- 
2.7.4

